### PR TITLE
Make emails confirmable

### PR DESCRIPTION
**Pull request by allenwq**
Add an extension to support `email.send_confirmation_instructions`
Declare confirmable in User and forward the logic to User::Email

### CODE REVIEW

File: config/locales/en/devise.yml [hunk 1]
```diff
@@ -19,6 +19,9 @@ en:
     mailer:
       confirmation_instructions:
         subject: 'Confirmation instructions'
+        welcome: 'Welcome %{email} !'
+        instruction: 'You can confirm your email through the link below:'
```
Comment (allenwq) [inline config/locales/en/devise.yml]: @lowjoel I know you have concerns about `:` at the end of the sentence, but in the devise mailer, they put these whole string in a paragraph, the final format is like:
```
<p>Welcome test@example.org !</p>

<p>You can confirm your email through the link below:</p>

<p><a href="http://localhost:3000/users/confirmation?confirmation_token=pdsMptRq23ymaxWg8uWR">Confirm my email</a></p>
```

**Comment by lowjoel** [comment]
I don't understand how Devise will allow these two models to coexist in the same application. Does that means that UserEmails can now be used as a User-like entity?

**Comment by allenwq** [comment]
Yup. And Devise won't check the detail of models, it considers everything as resources. 

Like in this case, after `email.send_confirmation_instructions`, devise will set the resource to email, and then I can get the resource(email) in views. But devise still expects an `User` in views, so that I need to provide my own email template.
https://github.com/plataformatec/devise/blob/master/lib/devise/mailers/helpers.rb#L20

**Comment by lowjoel** [comment]
Somehow I think this is more confusing than anything else. I would not expect Devise to be used like this.

**Comment by allenwq** [comment]
Do you have any better ideas ?

And I think we have discussed ? As said Devise does not support multiple emails. Also this is just the start, there is a lot of controller overridings which I'm about to push in later PRs.

**Comment by lowjoel** [comment]
Maybe we should switch to writing our own then... Otherwise we need to make our own confirmable Devise plugin.

If this were a separate gem, I think it's okay because people would expect to need to know Devise to work on it. But to be part of the main application would be a maintenance headache since it's another thing to worry about.

**Comment by allenwq** [comment]
...Okay, I'm not sure how much time it takes to write such a gem. But should be much harder since I need to design and make the APIs more general.

**Pull request closed without merging**